pkg/cloud/azure/services/resources/groups.go

```diff
@@ -17,11 +17,55 @@ limitations under the License.
 package resources
 
 import (
+	"fmt"
+
 	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2018-05-01/resources"
 	"github.com/Azure/go-autorest/autorest"
 	"github.com/Azure/go-autorest/autorest/to"
+	"k8s.io/klog"
 )
 
+// ReconcileResourceGroup reconciles the resource group of the given cluster.
+func (s *Service) ReconcileResourceGroup() (err error) {
```

Comment:
Not sure if there is a ResourceGroup equivalent in AWS. But I personally feel we shouldn't be using the Resource Group abstractions at all. (This is just my opinion though and not representative of Azure). As we are experimenting with the possibilities of VNet peering (using inter-resource-group Vnets), this could prove to be problematic. Definitely open to discussing this in detail in the future.

Reply:
@awesomenix FYI. Let me know If I may expressed anything incorrect here.

Reply:
AFAIK, AWS doesn't have an equivalent abstraction like resource group.
In the pre-refactor capz (`v0.1.0-alpha.1`), we're still creating resource groups; it's just happening in the deployment template as opposed to handled in the code.

I can see a future where we may allow multiple resource groups, say, to separate network from compute resources, but in the near-term, and definitely for this PR, we need to be able to handle them.

A huge benefit of having everything scoped to a single RG, especially at this stage of maturity for capz, is that cleanup is as simple as destroying the RG.